### DISCUSSION TITLE
Skip checkup fails

### DISCUSSION THREAD

**Discussion by Laurens-FLTRD** [skiptools/skip#303] [Q&A]
Hi, I'm setting up Skip and the checkup seems to keep failing. Woud appreciate some help and guidance on getting this resolved!

Output of `skip checkup --no-fail-fast`:
```
[✓] Skip version 1.2.10 (= 1.2.10)
[✓] macOS version 13.6.1 (> 13.5.0)
[✓] macOS architecture: ARM
[✓] Swift version 5.9.2 (> 5.9.0)
[✓] Xcode version 15.2 (> 15.0.0)
[✓] Xcode tools SDKs: 6
[✓] Homebrew version 4.4.12 (> 4.1.0)
[✓] Gradle version 8.11.1 (> 8.6.0)
[✓] Java version 23.0.1 (> 21.0.0)
[✓] Android Debug Bridge version 1.0.41 (> 1.0.40)
[✓] Android Studio version: 2024.2
[✓] Android tools SDKs: 2
[✓] Resolve dependencies (26.72s)
[✓] Build hello-skip (27.05s)
[✗] Test Swift (12.16s)
[✗] Test Kotlin (0.71s)
[✓] Archive iOS ipa (32.03s)
[✓] Assemble HelloSkip-release.ipa 63 KB
[✓] Verify HelloSkip-release.ipa 63 KB
[✗] Assembling Android apk (25.01s)
[✗] Verify HelloSkip-release.apk: The file “app-release-unsigned.apk” couldn’t be opened because there is no such file.
[✓] Check Swift Package (0.53s)
[✓] Check Skip Updates: 1.2.10
[✗] Skip 1.2.10 checkup failed with 4 errors
See https://skip.tools/docs/faq and https://community.skip.tools for help
See output log for error details: /tmp/skip-checkup-2024-12-17T19:53:25Z.txt
Error: 5 errors
```

Output log file
[skip-checkup-2024-12-17T19:53:25Z.txt](https://github.com/user-attachments/files/18170856/skip-checkup-2024-12-17T19.53.25Z.txt)

The output log file seems to hint at an error with XCTest, but I can't figure out what exactly is causing it.
```
error: terminated(1): /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/libexec/swift/pm/swiftpm-xctest-helper /private/var/folders/43/9bsp4hvn7b1f7m1d_vrj9z340000gn/T/F10D72CF-139B-4E0F-BE5D-89E6ACE8F991/hello-skip/.build/arm64-apple-macosx/debug/hello-skipPackageTests.xctest /var/folders/43/9bsp4hvn7b1f7m1d_vrj9z340000gn/T/TemporaryFile.lXhWvN output:
    error: unableToLoadBundle("/var/folders/43/9bsp4hvn7b1f7m1d_vrj9z340000gn/T/F10D72CF-139B-4E0F-BE5D-89E6ACE8F991/hello-skip/.build/arm64-apple-macosx/debug/hello-skipPackageTests.xctest")
```

**Comment by aabewhite** [comment]
Thanks for the report, and sorry for the trouble. We'll look into the logs. Until then, though, you might try simply ploughing ahead with the setup instructions. Try creating a project and building it in Xcode. Sometimes Xcode has different results than the command-line builds that the 'skip' tool performs

**Comment by marcprux** [comment]
I haven't seen that `unableToLoadBundle` error before. There is also a later error in the log file:

> Basics/Triple+Basics.swift:149: Fatal error: Cannot create dynamic libraries for os "ios".

We don't use macOS 13 day-to-day very much, but our CI does run on that version and it passes `skip checkup`, so I'm not sure what the problem would be.

I agree with Abe that you should try creating and building a project and trying to run it in Xcode. That might at least provide a more illuminating error message…

**Comment by Laurens-FLTRD** [comment]
Thanks for the suggestion Abe and Marc! I tried creating a project and building it in Xcode and it seems to work! Will be exploring further now if Skip fits my use-case.

**Comment by kfir26** [comment]
Hi skips :) 

my macbook info - 
 x86_64, on an Intel-based Mac.
macOS version (macOS 15.1.1)
homebrew installed already & android studio ladybug 2024 installed

im also getting this fail errors when running 
and the error caused by sysctl.proc_translated.



code that was written & output : 

skip checkup

output: 
[✓] Skip version 1.2.10 (= 1.2.10)
[✓] macOS version 15.1.1 (> 13.5.0)
[!] Error running sysctl (Intel)
[✗] Command failed with exit code 1
[✗] Skip 1.2.10 checkup failed with 1 error and 1 warning
See https://skip.tools/docs/faq and https://community.skip.tools for help
See output log for error details: /tmp/skip-checkup-2024-12-19T12:52:03Z.txt
Error: 2 errors

i'm not understand how to resolve this ...
please help me out..

**Reply by marcprux** [reply]
This is a known issue when running on an Intel machine. Can you try running `skip checkout --no-fail-fast` instead? If that completed, then you should be all set.

**Reply by aabewhite** [reply]
"checkup" rather than "checkout" :)

**Reply by kfir26** [reply]
Hi guys , tnx for the quick response 
Really appreciate it  :)

Soo  i've tried this code out its works and skipped the first step but got all those errors to be fixed ... 

Code: 
skip checkup --no-fail-fast

Error: 

[✓] Skip version 1.2.10 (= 1.2.10)
[✓] macOS version 15.1.1 (> 13.5.0)
[!] Error running sysctl (Intel)
[!] Swift version 5.2.4 (< 5.9.0)
[✗] Xcode version: error executing xcodebuild (install from: https://developer.apple.com/xcode/)
[✓] Xcode tools SDKs: 3
[✓] Homebrew version 4.4.12 (> 4.1.0)
[✓] Gradle version 8.11.1 (> 8.6.0)
[✓] Java version 23.0.1 (> 21.0.0)
[✓] Android Debug Bridge version 1.0.41 (> 1.0.40)
[✓] Android Studio version: 2024.2
[✓] Android tools SDKs: 1
[✗] Resolve dependencies (0.06s)
[✗] Build hello-skip (0.09s)
[✗] Test Swift (0.06s)
[✗] Test Kotlin (0.09s)
[✗] Archive iOS ipa (0.01s)
[✗] The file “Payload” doesn’t exist.
[✗] Skip 1.2.10 checkup failed with 7 errors and 2 warnings
See https://skip.tools/docs/faq and https://community.skip.tools for help
See output log for error details: /tmp/skip-checkup-2024-12-20T12:12:31Z.txt
Error: 8 errors

what need to be done to fixing it by my side
also need to be mentioned that my xcode version is the latest and updated form the app store .. soo im not really  know what this error
[✗] Xcode version: error executing xcodebuild (install from: https://developer.apple.com/xcode/)

**Reply by marcprux** [reply]
Can you run `xcodebuild -version` and post the results? It sounds like there might be an older version on your machine, since the Swift version referenced (5.2.4) is quite old. If you run `xcode-select --install`, does that change the `swift --version` output?

Also, can you post the log file referenced at the end of the `skip checkup` command?

**Reply by shoaib196** [reply]
`[✓] Skip version 1.2.25 (= 1.2.25)
[✓] macOS version 15.1.1 (> 13.5.0)
[✓] macOS architecture: ARM
[✓] Swift version 6.0.3 (> 5.9.0)
[✗] Xcode version: error executing xcodebuild (install from: https://developer.apple.com/xcode/)
[✗] xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance
[✗] Skip 1.2.25 checkup failed with 2 errors`



Just  Run sudo xcode-select -s /Applications/Xcode.app/Contents/Developer
 to resolve Xcode issues
